R/output_analysis.R

```diff
@@ -14,3 +14,29 @@ growth_rate <- function(out) {
   lambdas <- pops / c(NA, pops[1:(length(pops) - 1)])
   exp(mean(log(lambdas), na.rm = TRUE))
 }
+
+
+#' Calculate annual growth rate
+#'
+#' @param out Model output data frame
+#' @returns Numeric vector of length one representing the annual factor by
+#' which the total tick population changes. To use this function, it is best
+#' to run the model for at least three years.
+#'
+#' @examples
+#' out <- run(config_ex_1)
+#' annual_growth_rate(out)
+#'
+#' @export
+annual_growth_rate <- function(out) {
+  daily_data <- data.frame(
+    day = unique(out$day)
+  )
+  daily_data$totalpop <- sapply(daily_data$day, function(x) sum(out$pop[out$day == x]))
+  daily_data$yr <- ceiling(daily_data$day / 365)
+  years <- unique(daily_data$yr)
```

Comment:
I noticed in the function docs that you recommend at least 3 years of data. I thought we could easily enforce requiring at least 2 years of data, since you cannot calculate a year-over-year rate with just one year:
```suggestion
  years <- unique(daily_data$yr)
  if (length(years) < 2) {
    stop(
      "At least 2 years of data are required to calculate an annual growth rate",
      ", but the input data only has ", length(years), " years"
    )
  }
```